MidasCivil_Adapter/CRUD/Read/Loads/AreaDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,118 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using BH.oM.Adapters.MidasCivil;
+using BH.Engine.Adapter;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.Elements;
+using BH.Engine.Reflection;
+
+namespace BH.Adapter.MidasCivil
+{
+    public partial class MidasCivilAdapter
+    {
+        private List<ILoad> ReadAreaDifferentialTemperatureLoads(List<string> ids = null)
+        {
+            List<ILoad> bhomAreaDifferentialTemperatureLoads = new List<ILoad>();
+
+            List<Loadcase> bhomLoadcases = ReadLoadcases();
+            Dictionary<string, Loadcase> loadcaseDictionary = bhomLoadcases.ToDictionary(
+                        x => x.Name);
+
+            string[] loadcaseFolders = Directory.GetDirectories(m_directory + "\\TextFiles");
+
+            foreach (string loadcaseFolder in loadcaseFolders)
+            {
+                string loadcase = Path.GetFileName(loadcaseFolder);
+                List<string> areaDifferentialTemperatureLoadText = GetSectionText(loadcase + "\\THERGRAD");
+                List<string> areaUniformTemperatureLoadText = GetSectionText(loadcase + "\\ELTEMPER");
+                List<string> feMeshComparison = new List<string>();
+                List<string> loadedFEMeshes = new List<string>();
+
+                if (areaDifferentialTemperatureLoadText.Count != 0)
+                {
+                    List<string> areaDifferentialTempatureElements = areaDifferentialTemperatureLoadText.Select(x => x.Split(',')[0]).ToList();
+                    List<string> areaUniformTempatureElements = areaUniformTemperatureLoadText.Select(x => x.Split(',')[0]).ToList();
+                    for (int i = 0; i < areaDifferentialTempatureElements.Count; i++)
+                    {
+                        List<string> differentialDelimitted = areaDifferentialTemperatureLoadText[i].Split(',').ToList();
+                        loadedFEMeshes.Add(areaDifferentialTempatureElements[i].Trim());
+                        differentialDelimitted.RemoveAt(0);
+                        if (areaUniformTempatureElements.Count != 0)
+                        {
+                            if (areaDifferentialTempatureElements[i] == areaUniformTempatureElements[i])
+                            {
+                                List<string> globalDelimitted = areaUniformTemperatureLoadText[i].Split(',').ToList();
+                                differentialDelimitted.Add(globalDelimitted[1].Trim());
```

Comment:
Can you store the global temperature offset in a separate `List<double>` please and feed that in to the `Convert` method.

By adding it to the end of an existing `string` is not immediately clear what it represents or where it has come from. You can add an additional input to the `Convert` method.